test/unit/lib/runner/writers/action-json-writer-tests.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const mockery = require('mockery');
+const sinon = require('sinon');
+const expect = require('chai').expect;
+
+describe('lib/runner/test-runner/action-json-writer.js', function() {
+  describe('writeActionReport', function() {
+    let fs;
+    let path;
+    let writeActionReportToDisk;
+    let configHandler;
+
+    let report;
+
+    beforeEach(function() {
+      mockery.enable({useCleanCache: true});
+      mockery.registerAllowable('../../../../../lib/runner/writers/action-json-writer.js');
+
+      path = {
+        resolve: sinon.stub(),
+      };
+
+      report = {
+        testReports: [
+          {
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'modal',
+                      action: 'click',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+          {
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
```

Comment:
these tests should have different names perhaps so tests checking testName are more powerful, and we know they are getting the test report we expect them to get